extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/theta/SketchAggregator.java

```diff
@@ -133,4 +174,34 @@ static void updateUnion(Union union, Object update)
       throw new ISE("Illegal type received while theta sketch merging [%s]", update.getClass());
     }
   }
+
+  /**
+   * Gets the initial size of this aggregator in bytes.
+   */
+  public long getInitialSizeBytes()
+  {
+    // SketchAggregator has 3 references and an int
+    // UnionImpl has a reference, a short, a long, a boolean
+    long sizeOfReferences = 3L * Long.BYTES + Integer.BYTES
+                            + Long.BYTES + Short.BYTES + Long.BYTES + 1;
+    if (sketchField == null) {
+      return sizeOfReferences;
+    }
+
+    // Initialize the sketch if not already initialized
+    if (union == null) {
+      initUnion();
```

Comment:
This feels a little bit wrong to me to init on the initial sizing.  If it starts out `null`, then its size is `0`.  It seems like we should either do this lazily, OR we should initialize the union on construction.

Reply:
Re-structured the code to initialize lazily and account for the size only when initializing.